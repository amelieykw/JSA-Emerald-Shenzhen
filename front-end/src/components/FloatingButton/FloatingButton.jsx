import React from 'react'
import { Text } from 'react-native'
import { Button, Icon } from 'native-base'
import PropTypes from 'prop-types'
import style from './FloatingButtonStyle'

function FloatingButton({ handlePress, buttonStyle, text, icon }) {
  return (
    <Button
      onPress={handlePress}
      rounded
      style={{
        ...style.button,
        width: buttonStyle.width ? buttonStyle.width : style.button.width,
        backgroundColor: buttonStyle.backgroundColor
          ? buttonStyle.backgroundColor
          : style.button.backgroundColor
      }}
    >
      <Icon style={style.buttonIcon} type={icon.type} name={icon.name} />
      <Text style={style.buttonTitle}>{text}</Text>
    </Button>
  )
}
FloatingButton.defaultProps = {
  buttonStyle: {}
}
FloatingButton.propTypes = {
  handlePress: PropTypes.func.isRequired,
  icon: PropTypes.objectOf(PropTypes.any).isRequired,
  text: PropTypes.string.isRequired,
  buttonStyle: PropTypes.objectOf(PropTypes.any)
}
export default FloatingButton
