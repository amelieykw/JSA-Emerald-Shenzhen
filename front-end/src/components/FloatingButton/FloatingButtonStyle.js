import { StyleSheet } from 'react-native'

const style = StyleSheet.create({
  button: {
    backgroundColor: '#F66F43',
    width: 110,
    height: 60,
    borderRadius: 35,
    shadowColor: 'black',
    shadowOffset: {
      width: 0,
      height: 3
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,
    elevation: 5,
    marginRight: 10,
    position: 'absolute',
    bottom: 10,
    right: 5
  },
  buttonTitle: {
    left: -5,
    fontSize: 20,
    color: 'white'
  },
  buttonIcon: {
    fontSize: 25,
    left: 0
  }
})

export default style
