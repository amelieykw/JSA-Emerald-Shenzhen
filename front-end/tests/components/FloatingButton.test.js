import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import FloatingButton from '../../src/components/FloatingButton/FloatingButton'

Enzyme.configure({ adapter: new Adapter() })

const handlePress = () => {
  return {}
}

const icon = { type: 'MaterialIcons', name: 'done' }

describe('test NewLoan View', () => {
  it('testing FloatingButton', () => {
    const warpper = shallow(<FloatingButton handlePress={handlePress} icon={icon} text="test" />)
    expect(warpper).toMatchSnapshot()
  })
})
