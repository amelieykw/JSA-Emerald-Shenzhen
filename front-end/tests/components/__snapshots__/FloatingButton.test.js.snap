// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test NewLoan View testing FloatingButton 1`] = `ShallowWrapper {}`;
